cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SystemProgramManagementService.java

```diff
@@ -133,7 +133,7 @@ private void startPrograms(Map<ProgramId, Arguments> enabledProgramsMap) {
   private void stopProgram(ProgramRunId programRunId) {
     LOG.debug("Stopping program run {} ", programRunId);
     try {
-      programLifecycleService.stop(programRunId.getParent(), programRunId.getRun());
+      programLifecycleService.stop(programRunId.getParent(), programRunId.getRun(), 30);
```

Comment:
Why specifically 30 secs?

Reply:
hm this should be integer max actually, will change